packages/jest-mock/src/index.js

```diff
@@ -37,6 +37,8 @@ type MockFunctionConfig = {
 
 const MOCK_CONSTRUCTOR_NAME = 'mockConstructor';
 
+const FUNCTION_NAME_RESERVED_PATTERN = /[\s!-\/:-@\[-`{-~]/g;
```

Comment:
Does it needs to contain a `global` flag? This makes this regex stateful, which may result in unexpected behaviors like:

<img width="372" alt="screen shot 2017-09-12 at 08 46 22" src="https://user-images.githubusercontent.com/5106466/30311740-04232e6a-9797-11e7-8f25-6687d869af20.png">

Reply:
Ohh, yeah, let's fix that. Good find.

Reply:
The call to `.test` doesn't need the flag, but the call to `.replace` does.

Reply:
I'll put up another PR to fix this

Reply:
Oh I see it's already fixed in master. Thanks for spotting this @thymikee !